From 6e3a1f46c2499e13e634354af60a6d30088ffe10 Mon Sep 17 00:00:00 2001
From: Ben Horgan <ben.horgan@arm.com>
Date: Mon, 4 Sep 2023 14:22:10 +0100
Subject: [PATCH 18/18] tc2: fpga: Enable gpu/dpu scmi power domain and also
 gpu perf domain

Same change for fpga as done for fvp previously with the same
commit headline. Power domain ids are different as they change with
the number of cpus.

Signed-off-by: Ben Horgan <ben.horgan@arm.com>
---
 fdts/tc_fpga.dts | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/fdts/tc_fpga.dts b/fdts/tc_fpga.dts
index 611d06653..b4eb9d821 100644
--- a/fdts/tc_fpga.dts
+++ b/fdts/tc_fpga.dts
@@ -426,6 +426,11 @@
 		#address-cells = <1>;
 		#size-cells = <0>;
 
+		scmi_devpd: protocol@11 {
+			reg = <0x11>;
+			#power-domain-cells = <1>;
+		};
+
 		scmi_dvfs: protocol@13 {
 			reg = <0x13>;
 			#clock-cells = <1>;
@@ -567,6 +572,7 @@
 		interrupt-names = "DPU";
 		clocks = <&dpu_aclk>;
 		iommus = <&smmu_700 0x100>;
+		power-domains = <&scmi_devpd 16>;
 
 		clock-names = "aclk";
 		pl0: pipeline@0 {
@@ -643,14 +649,12 @@
 		reg = <0x0 0x2d000000 0x0 0x200000>;
 		interrupts = <0 66 4>, <0 67 4>, <0 65 4>;
 		interrupt-names = "JOB", "MMU", "GPU";
-		clocks = <&gpu_clk>, <&gpu_core_clk>;
-		clock-names = "clk_mali", "shadercores";
 		power_policy = "always_on";
+		clocks = <&gpu_core_clk>;
+		clock-names = "shadercores";
+		power-domains = <&scmi_devpd 15>;
+		scmi-perf-domain = <3>;
 		iommus = <&smmu_700 0x200>;
 		kvm,protected-device;
-		operating-points = <
-			/* KHz uV */
-			50000 820000
-		>;
 	};
 };
-- 
2.25.1

